pytorch_lightning/trainer/training_loop.py

```diff
@@ -515,8 +516,22 @@ def run_training_batch(self, batch, batch_idx):
         for split_idx, split_batch in enumerate(splits):
             self.split_idx = split_idx
 
+            def get_optimizers_iterable():
+                if not self.optimizer_frequencies:
+                    return enumerate(self.optimizers)
+
+                optimizer_freq_cumsum = np.cumsum(self.optimizer_frequencies)
+                optimizers_loop_length = optimizer_freq_cumsum[-1]
+                current_place_in_loop = self.total_batch_idx % optimizers_loop_length
+
+                # find optimzier index by looking for the first {item > current_place} in the cumsum list
+                for opt_idx, v in enumerate(optimizer_freq_cumsum):
+                    if v > current_place_in_loop:
+                        # return an iterable list of one tuple
+                        return [(opt_idx, self.optimizers[opt_idx])]
```

Comment:
really `return` and not `yield`? because `return` terminates the function so there would not be a reason for the loop...

Reply:
It is exactly what I want to achieve - I want the function to terminate.
The whole idea of this loop is just to find the first element in `optimizer_freq_cumsum` that is bigger than `current_place_in_loop` and retrieve its index.
Maybe there is a prettier way to do it using `np.argmax()`?
This is what we should consider...

Reply:
I believe I should implement it using np.argmax() like that:
```py
opt_idx = np.argmax(optimizer_freq_cumsum > current_place_in_loop)
return [(opt_idx, self.optimizers[opt_idx])]
```

Reply:
I've implemented using np.argmax(), see the comments below, let's discuss there.